### PR TITLE
add rcl_action_client_options when creating action client.

### PR DESCRIPTION
**Pull request by fujitatomoya**
fix https://github.com/ros2/rclcpp/issues/1118.

### CODE REVIEW

**Comment by fujitatomoya** [comment]
I guess that it needs to be rebased.

**Comment by fujitatomoya** [comment]
@ros-pull-request-builder retest this please

**Comment by fujitatomoya** [comment]
source build with https://github.com/ros2/ros2/commit/4eece71bc42f9b28e1c21633d270d908bb508ca7,

i do not see the following build error reported by CI,
http://build.ros2.org/job/Fpr__rclcpp__ubuntu_focal_amd64/80/console#console-section-15
```
08:03:23 /tmp/ws/src/rclcpp/rclcpp/src/rclcpp/context.cpp: In function ‘void rclcpp_logging_output_handler(const rcutils_log_location_t*, int, const char*, rcutils_time_point_value_t, const char*, __va_list_tag (*)[1])’:
08:03:23 /tmp/ws/src/rclcpp/rclcpp/src/rclcpp/context.cpp:66:12: error: ‘rcl_logging_multiple_output_handler’ was not declared in this scope; did you mean ‘rcutils_logging_console_output_handler’?
08:03:23    66 |     return rcl_logging_multiple_output_handler(
08:03:23       |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
08:03:23       |            rcutils_logging_console_output_handler
08:03:23 /tmp/ws/src/rclcpp/rclcpp/src/rclcpp/context.cpp:67:56: error: return-statement with a value, in function returning ‘void’ [-fpermissive]
08:03:23    67 |       location, severity, name, timestamp, format, args);
08:03:23       |                                                        ^
08:03:23 /tmp/ws/src/rclcpp/rclcpp/src/rclcpp/context.cpp: In member function ‘virtual void rclcpp::Context::init(int, const char* const*, const rclcpp::InitOptions&)’:
08:03:23 /tmp/ws/src/rclcpp/rclcpp/src/rclcpp/context.cpp:146:13: error: ‘rcl_logging_configure_with_output_handler’ was not declared in this scope; did you mean ‘rcutils_logging_console_output_handler’?
08:03:23   146 |       ret = rcl_logging_configure_with_output_handler(
08:03:23       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
08:03:23       |             rcutils_logging_console_output_handler
08:03:23 make[2]: *** [CMakeFiles/rclcpp.dir/build.make:220: CMakeFiles/rclcpp.dir/src/rclcpp/context.cpp.o] Error 1
08:03:23 make[1]: *** [CMakeFiles/Makefile2:105: CMakeFiles/rclcpp.dir/all] Error 2
08:03:23 make: *** [Makefile:141: all] Error 2
08:03:23 ---
```

i do not think this is related to my fix.

**Comment by fujitatomoya** [comment]
@hidmic 

friendly ping.

File: rclcpp_action/include/rclcpp_action/client.hpp [hunk 1]
```diff
@@ -130,7 +130,7 @@ class ClientBase : public rclcpp::Waitable
     rclcpp::node_interfaces::NodeLoggingInterface::SharedPtr node_logging,
     const std::string & action_name,
     const rosidl_action_type_support_t * type_support,
-    const rcl_action_client_options_t & options);
+    const rcl_action_client_options_t & options = rcl_action_client_get_default_options());
```
Comment (wjwwood) [inline rclcpp_action/include/rclcpp_action/client.hpp:133]: This change seems unnecessary to me. I think we should not make this optional.

Comment (fujitatomoya) [reply rclcpp_action/include/rclcpp_action/client.hpp]: agree, thanks for the catch! fixed.

**Comment by jacobperron** [comment]
> it would be better to have a C++ based option structure.

+1
If we do this we should also add a C++ structure for the server options. This sounds like a good follow-up PR.

**Comment by fujitatomoya** [comment]
@jacobperron @hidmic @wjwwood 

merge conflicts are resolved, could we have a go for merge?

**Comment by jacobperron** [comment]
CI for packages above rclcpp_action:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/11146/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/6436/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9091/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11064/)

**Comment by jacobperron** [comment]
@fujitatomoya Thanks for the patch!

**Pull request merged**